ui/src/components/select/QasSelect.vue

```diff
@@ -30,6 +30,32 @@
       </qas-badge>
     </template>
 
+    <template v-if="useCustomOption" #option="scope">
+      <q-item v-bind="scope.itemProps" class="qas-select__option">
+        <q-item-section>
+          <div class="items-center q-gutter-x-sm row">
+            <q-item-label>{{ scope.opt.label }}</q-item-label>
+
+            <div v-for="(badge, index) in getFilteredBadgeList(scope.opt)" :key="index">
+              <qas-badge v-if="canShowBadge(badge)" v-bind="getBadgeProps(badge)" />
+            </div>
+          </div>
+
+          <div v-if="scope.opt.caption">
+            <div v-if="isCaptionArray(scope.opt.caption)" class="items-center q-col-gutter-x-sm row">
+              <q-item-label v-for="(caption, index) in scope.opt.caption" :key="index" caption class="items-center q-mt-xs row">
+                <div>{{ caption }}</div>
+
+                <q-separator v-if="!isLastCaption({ caption: scope.opt.caption, index })" class="q-ml-sm" vertical />
+              </q-item-label>
```

Comment:
pq não transforma sempre o caption em array mesmo quando for string, assim não precisa ter essa repetição de código!!